send-transaction-service/src/tpu_info.rs

```diff
@@ -1,15 +1,15 @@
-use std::net::SocketAddr;
+use {solana_client::connection_cache::Protocol, std::net::SocketAddr};
 
 pub trait TpuInfo {
-    fn refresh_recent_peers(&mut self);
+    fn refresh_recent_peers(&mut self, protocol: Protocol);
     fn get_leader_tpus(&self, max_count: u64) -> Vec<&SocketAddr>;
```

Comment:
This api doesn't stop someone from `refresh_recent_peers(Protocol::UDP)` and then expecting  `get_leader_tpus` to return the QUIC port or vice-versa.
Can we instead move the `: Protocol` to `get_leader_tpus`?

Reply:
However it is refresh_recent_peers actually loading the addresses into the cache -- get_leader_tpus just consumes the cache. I would like to avoid the port offset logic here again. I think we can leave the interface unchanged and let ClusterTpuInfo's constructor decides it.

Reply:
I was thinking `ClusterTpuInfo.recent_peers` would cache both QUIC and UDP ports, and then `get_leader_tpus` can return the correct one based on protocol argument.